### PR TITLE
Make iterations in `SugiyamaAlgorithm` configurable

### PR DESCRIPTION
**Pull request by JulianBissekkou**

### CODE REVIEW

**Comment by JulianBissekkou** [comment]
@nabil6391 

This PR introduces some smaller changes that help us increase performance. The first one is making the iterations in `nodeOrdering` configurable since for all our graphs one iteration is enough. This change is non breaking since i am keeping the previous value of `10`.

I also avoided recalculating some stuff and did some smaller optimizations in `crossingb` which is called very often. The increased performance just a little bit. As you said the bigger fix would be to optimize `assignX` and `transpose`.
This is still open. See #56 .

I also did some smaller code style changes, but i can revert them if you like.

Thanks for taking the time!

File: lib/layered/SugiyamaAlgorithm.dart [hunk 1]
```diff
@@ -863,7 +858,7 @@ class SugiyamaAlgorithm extends Algorithm {
     nodeOrdering(); //expensive operation
     coordinateAssignment(); //expensive operation
     // shiftCoordinates(shiftX, shiftY);
-    final graphSize = calculateGraphSize(this.graph);
+    //final graphSize = calculateGraphSize(this.graph);
```
Comment (JulianBissekkou) [inline lib/layered/SugiyamaAlgorithm.dart:861]: This was unused so i commented it out 👍🏽

**Comment by nabil6391** [comment]
Thanks for this. I will create some tests and have a look at this on the weekend

**Comment by nabil6391** [comment]
Hi @JulianBissekkou  , I have added two tests in the repo, can you please try to run them and see if they pass. Thanks 

Other than that LGTM

**Comment by JulianBissekkou** [comment]
@nabil6391 Tests are passing! I also created a workflow file that runs tests in master and on every pr. I think you have to enable github actions in the project otherwise I can not test if the workflow works :)

**Pull request merged**